doc/source/conf.py

```diff
@@ -26,12 +26,33 @@
 # Add any Sphinx extension module names here, as strings. They can be extensions
 # coming with Sphinx (named 'sphinx.ext.*') or your custom ones.
 extensions = ['IPython.sphinxext.ipython_console_highlighting',
-              'IPython.sphinxext.ipython_directive']
+              'IPython.sphinxext.ipython_directive',
+              'sphinx_gallery.gen_gallery',
+              'sphinx.ext.autosummary', 'sphinx.ext.intersphinx',
+]
+
+def setup(app):
+    app.add_stylesheet('custom.css')  # may also be an URL
 
 # Add any paths that contain templates here, relative to this directory.
 
 templates_path = ['_templates']
 
+autosummary_generate = True
+
+# Sphinx gallery configuration
+sphinx_gallery_conf = {
+    'examples_dirs': ['../../gallery'],
+    'filename_pattern': '^((?!sgskip).)*$',
+    'gallery_dirs': ['gallery'],
+    'doc_module': ('geopandas',),
+    'reference_url': {'matplotlib': 'http://matplotlib.org',
+                      'numpy': 'http://docs.scipy.org/doc/numpy/reference',
+                      'scipy': 'http://docs.scipy.org/doc/scipy/reference',
+                      'geopandas': None},
+    'backreferences_dir': 'api/_as_gen',  # Not currently used but kept for future
```

Comment:
I'd say it's technically used but not super utilized. It's meant for API pages (e.g., on the API page for `from_file` it'll list all the sphinx-gallery examples that use that method/function/etc)